src/simulation/elements/FIRE.cpp

```diff
@@ -127,11 +127,17 @@ int Element_FIRE_update(UPDATE_FUNC_ARGS)
 					}
 					else if (t==PT_LAVA)
 					{
-						if (parts[i].ctype == PT_IRON && RNG::Ref().chance(1, 500)) {
+						if (parts[i].ctype == PT_IRON && RNG::Ref().chance(1, 500))
+						{
 							parts[i].ctype = PT_METL;
 							sim->kill_part(ID(r));
-							continue;
+						} 
+						else if ((parts[i].ctype == PT_STNE || parts[i].ctype == PT_NONE) && RNG::Ref().chance(1, 60))
+						{
+							parts[i].ctype = PT_SLCN;
+							sim->kill_part(ID(r));
 						}
+            continue;
```

Comment:
Funky indentation and bracket placement is a bit inconsistent. This applies elsewhere also but I couldn't be bothered to flag everything and it's not that big a deal.